### DISCUSSION TITLE
Add reviewers to Azure DevOps pull request

### DISCUSSION THREAD

**Discussion by snailcatcher** [renovatebot/renovate#17728] [Request Help] [answered]
Hi there,
I've tried to add a reviewer to my pull requests in Azure DevOps generated by renovate.
Since the documentation of this feature is not very detailed, I was forced to use the try-and-error approach.
I've tried all of the following lines in my config.js file but none of them actually did anything:

```
reviewers: ["team:teamname"]
reviewers: ["teamname"]
reviewers: ["[projectname]\\teamname"]
reviewers: ["my.mail@address.com"]
reviewers: ["my-user-name"]
additionalReviewers: ["team:teamname"]
additionalReviewers: ["teamname"]
```

My wish is to add the team as an optional reviewer to each pull request.

I already looked into the discussions:
#17659
#13594

Is it possible to update an already created pull request with new reviewers?

sincerely snailcatcher

**Comment by rarkins** [comment]
Reviewers are only added at the time of PR creation. I'm not sure if team reviews are a supported feature

**Reply by snailcatcher** [reply]
Okay I can life with this, so I have to add the reviewers as single users.
But what is the correct term for a user, because non of the above seems to work for me?

**Reply by rarkins** [reply]
@sch1ldkr0ete you were asking about required reviewers recently. Did you get regular reviewers to work?

**Reply by rarkins** [reply]
Here is a past issue on the topic: https://github.com/renovatebot/renovate/issues/13030

**Reply by sch1ldkr0ete** [reply]
Yes. Even team reviewers are working for me with `reviewers: ["teamname"]`. I think for getting adding reviewers to work, it is important to give the build user the permission to contribute to pull requests as it is described [here](https://marketplace.visualstudio.com/items?itemName=ShaykiAbramczyk.CreatePullRequest#for-azure-devops-repository).

**Reply by snailcatcher** [reply]
Yes, it is written in the documentation [here](https://docs.renovatebot.com/modules/platform/azure/#using-a-single-pipeline-to-update-multiple-repositories).

**Comment by snailcatcher** [comment, marked as answer]
So after some try-and-error and looking into the renovate code I found the solution.
It is `reviewers: ["TEAM NAME WITH ALL SPACES AND STUFF"]`
I think i did a spelling mistake or something like that in my first attempt.
@rarkins thx. your link in the issue helped a lot to find the correct spot in the code.

And if someone has the same issue:
1. It only works on new PRs, not on updates
2. Teams and users are handled exactly the same. Use their *displayName* oder *uniqueName* (email)

Helpful commands for finding issues:
1. *az devops team list* (shows all teams)
2. *az devops team list-member --organization **ORGANIZATION_URL** --project **PROJECT_NAME** --team "**TEAM_NAME**"* (shows all team members and their properties)
